deps/rabbit/src/rabbit_classic_queue.erl

```diff
@@ -519,22 +523,49 @@ update_msg_status(down, Pid, #msg_status{pending = P} = S) ->
 confirm_to_sender(Pid, QName, MsgSeqNos) ->
     %% the stream queue included the queue type refactoring and thus requires
     %% a different message format
-    Evt = case rabbit_feature_flags:is_enabled(stream_queue) of
-              true ->
-                  {queue_event, QName, {confirm, MsgSeqNos, self()}};
-              false ->
-                  {confirm, MsgSeqNos, self()}
-          end,
-    gen_server2:cast(Pid, Evt).
+    case rabbit_queue_type:is_supported() of
+        true ->
+            gen_server:cast(Pid,
+                            {queue_event, QName,
+                             {confirm, MsgSeqNos, self()}});
+        false ->
+            gen_server2:cast(Pid, {confirm, MsgSeqNos, self()})
+    end.
 
 send_rejection(Pid, QName, MsgSeqNo) ->
-    case rabbit_feature_flags:is_enabled(stream_queue) of
+    case rabbit_queue_type:is_supported() of
         true ->
-            gen_server2:cast(Pid, {queue_event, QName,
-                                   {reject_publish, MsgSeqNo, self()}});
+            gen_server:cast(Pid, {queue_event, QName,
+                                  {reject_publish, MsgSeqNo, self()}});
         false ->
             gen_server2:cast(Pid, {reject_publish, MsgSeqNo, self()})
     end.
 
-send_queue_event(Pid, QName, Evt) ->
-    gen_server2:cast(Pid, {queue_event, QName, Evt}).
+deliver_to_consumer(Pid, QName, CTag, AckRequired, Message) ->
+    case rabbit_queue_type:is_supported() of
+        true ->
+            Deliver = {deliver, CTag, AckRequired, [Message]},
+            Evt = {queue_event, QName, Deliver},
+            gen_server:cast(Pid, Evt);
```

Comment:
Are this message and the messages in the functions below specific to the interaction between a classic queue and a channel?

In other words, it this the right place to encode/format the message?

Reply:
yes the compatibility code is specific to classic queues, the queue_event format is generic to any queue type. Arguably we could forward to another function in, say, `rabbit_queue_type` but it would still only be used by classic queues as quorum queues do it differently and streams don't send these messages at all.

Reply:
Ok, thanks!